mesonbuild/cmake/client.py

```diff
@@ -103,19 +105,21 @@ def log(self) -> None:
         pass
 
 class Progress(MessageBase):
-    def __init__(self, cookie: str):
+    def __init__(self, cookie: str) -> None:
         super().__init__('progress', cookie)
 
     def log(self) -> None:
         pass
 
 class MessageHello(MessageBase):
-    def __init__(self, supported_protocol_versions: T.List[dict]):
+    def __init__(self, supported_protocol_versions: T.List[T.Dict[str, int]]) -> None:
```

Comment:
Could you add a TODO comment here about using a typed dict?

Reply:
This data directly comes from CMake and is never really checked in any way (otherwise there would have to be *many* checks and the entire CMake code would be twice as long). We are trusting that CMake always returns the right types here.

I guess it won't hurt, but I am also not sure that we gain anything from it either.